pkg/curl/data_request.go

```diff
@@ -121,6 +127,23 @@ func (d *curlRequestDataSource) Read(ctx context.Context, req datasource.ReadReq
 		return
 	}
 
+	headers := state.Headers.ValueString()
+
+	if strings.Trim(headers, " \t\r\n") != "" {
+		parsedHeaders, err := parseJSON(headers)
+		if err != nil {
+			resp.Diagnostics.AddError(
+				"Unable to set http headers",
+				err.Error(),
+			)
+			return
+		}
+
+		for header, value := range parsedHeaders {
+			httpRequest.Header.Set(header, value.(string))
+		}
+	}
```

Comment:
Change the inpyt to a map to simplify this entire part.

Reply:
I did make the changes, could you please re-review once again?